cli/run/loadConfigFile.ts

```diff
@@ -173,3 +204,55 @@ async function getConfigList(configFileExport: any, commandOptions: any): Promis
 	}
 	return Array.isArray(config) ? config : [config];
 }
+
+function tryResolveIdFromLoadedPlugins(
+	warnings: BatchWarnings,
+	id: string,
+	importer: string | undefined,
+	plugins: readonly Plugin<any>[]
+) {
+	for (const { resolveId, name } of plugins) {
+		const handler = typeof resolveId === 'function' ? resolveId : resolveId?.handler;
+		if (handler) {
+			// Since we don't support promises (because externality check is not defined as supporting them), we don't support plugin context either
+			// There probably is some way to do that (maybe save each plugin context by wrapping its buildStart hook?), but for now, let's just omit that.
+			let resolveIdResult: ResolveIdResult | Promise<ResolveIdResult>;
+			try {
+				resolveIdResult = handler.bind(undefined!)(id, importer, {
```

Comment:
Is there any feasible way to provide `PluginContext` here? Right now this works only because TS plugin `resolveId` does not use `this` in any way.

Reply:
You're right, it wasn't good idea to execute resolveId hooks here. I went with the suggested approach of just moving pre-existing externality check to be run as last thing, so this got fixed through that as well. :)